test/core/external/lib/mockZeroExOrderDataHandlerLibrary.spec.ts

```diff
@@ -0,0 +1,190 @@
+      { from: ownerAccount, gas: DEFAULT_GAS },
+    );
+
+    zeroExExchangeWrapper = new MockZeroExOrderDataHandlerLibraryContract(
+      web3.eth.contract(zeroExExchangeWrapperInstance.abi).at(zeroExExchangeWrapperInstance.address),
+      { from: ownerAccount },
+    );
+  });
+
+  describe("#parseOrderDataHeader", async () => {
+    // Header Subject Data
+    let signatureLength: UInt;
+    let zeroExOrderLength: BigNumber;
+    let makerAssetDataLength: BigNumber;
+    let takerAssetDataLength: BigNumber;
+
+
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = generateStandardZeroExOrderBytesArray(
+        zeroExOrder,
+        signature,
+        fillAmount,
+      );
+
+      const zeroExOrderBuffer = bufferZeroExOrder(zeroExOrder);
+      zeroExOrderLength = getZeroExOrderLengthFromBuffer(zeroExOrderBuffer);
+
+      signatureLength = new BigNumber(signature.length);
+      makerAssetDataLength = new BigNumber(makerAssetData.length);
+      takerAssetDataLength = new BigNumber(takerAssetData.length);
+    });
+
+    it("should correctly parse the order data header", async () => {
+      const result = await zeroExExchangeWrapper.parseOrderDataHeader.callAsync(subjectOrderData);
+
+      expect(result[0]).to.bignumber.equal(signatureLength);
+      expect(result[1]).to.bignumber.equal(zeroExOrderLength);
+      expect(result[2]).to.bignumber.equal(makerAssetDataLength);
+      expect(result[3]).to.bignumber.equal(takerAssetDataLength);
+    });
+  });
+
+  describe("#parseFillAmount", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = generateStandardZeroExOrderBytesArray(
+        zeroExOrder,
+        signature,
+        fillAmount,
+      );
+    });
+
+    it("correctly parse the fill amount", async () => {
+      const result = await zeroExExchangeWrapper.parseFillAmount.callAsync(subjectOrderData);
+      expect(result).to.be.bignumber.equal(fillAmount);
+    });
+  });
+
+  describe("#parseSignature", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = generateStandardZeroExOrderBytesArray(
+        zeroExOrder,
+        signature,
+        fillAmount,
+      );
+    });
+
+    it("should correctly parse the signature", async () => {
+      const result = await zeroExExchangeWrapper.parseSignature.callAsync(subjectOrderData);
+      expect(web3.toAscii(result)).to.equal(signature);
+    });
+  });
+
+  describe("#parseZeroExOrderData", async () => {
+    let subjectOrderData: Bytes32;
+
+    beforeEach(async () => {
+      subjectOrderData = generateStandardZeroExOrderBytesArray(
+        zeroExOrder,
+        signature,
+        fillAmount,
+      );
+    });
+
+    it("should correctly parse the zeroEx order", async () => {
+      const result = await zeroExExchangeWrapper.parseZeroExOrderData.callAsync(subjectOrderData);
```

Comment:
Seems like you don't need this result variable. Assign to the expected outputs. Maybe split up parsing the return variables and do the correct expectations after each one. Or better yet, add a separate spec for each.